app/forms/agreement_form.rb

```diff
@@ -7,7 +7,7 @@ class AgreementForm < Reform::Form
   property :agreement_file2, virtual: true
 
   validates :title, presence: true
-  validates :source_id, format: {with: /\A\w+:\w+\z/,
+  validates :source_id, format: {with: Regexp.new(Settings.source_id_regex),
```

Comment:
I'd ordinarily ask for a test here but this change makes the validation less strict (because `.` covers more characters than `\w`) so I don't think it is worth the time.